modules/flyte/iam.tf

```diff
@@ -0,0 +1,102 @@
+resource "aws_iam_role" "flyte_controlplane" {
+  name = "${local.deploy_id}-flyte-controlplane"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Effect = "Allow"
+        Principal = {
+          Federated = local.oidc_provider_arn
+        }
+        Condition : {
+          StringEquals : {
+            "${trimprefix(local.oidc_provider_url, "https://")}:aud" : "sts.amazonaws.com",
+            "${trimprefix(local.oidc_provider_url, "https://")}:sub" : [
+              "system:serviceaccount:${var.platform_namespace}:flyteadmin",
```

Comment:
can we not hardcode the service accounts.

Reply:
following the structure in https://github.com/dominodatalab/terraform-aws-eks/blob/main/modules/irsa/variables.tf#L70

Reply:
Can you explain the value of making these configurable? These service account names are defined by Flyte and not likely to change between different installs

Reply:
yes, the value lies in the value originating outside the module in a place that it propagates to terraform and also to the installer-> charts. Failure to do so will cause a disconnect between the module and the installer given that they do not inherit the value.

Reply:
Another approach is to actually use the variable :
https://github.com/dominodatalab/terraform-aws-eks/blob/63ada53bd6a6d380c45384ab86375b5fda4133b0/modules/flyte/variables.tf#L55
then set it as part of the outputs and have the installer pass it to the flyte chart

Reply:
I think I went with this "other approach" that you are suggesting...is there anything else you wanted me to do here?

Reply:
I dont see the service accounts as part of the outputs, so I am assuming that the values will be passed from the deployer and populated to the installer.